src/network/networkd-link.c

```diff
@@ -1338,6 +1338,60 @@ static int link_set_bridge(Link *link) {
         return r;
 }
 
+static int link_bond_set_(Link *link) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        int r;
+
+        assert(link);
+        assert(link->network);
+
+        r = sd_rtnl_message_new_link(link->manager->rtnl, &req, RTM_NEWLINK, link->network->bond->ifindex);
+        if (r < 0)
+                log_link_error_errno(link, r, "Could not allocate RTM_SETLINK message: %m");
```

Comment:
if this fails then we proceed here with req being NULL. this should fail the whole function, no?